### PR TITLE
[NO JIRA] Update PR template

### PR DESCRIPTION
**Pull request by anambl**
Remember to include the following changes:

- [ ] `README.md` (If you have created a new component)
- [ ] Component `README.md`
- [ ] Tests
- [ ] Storybook examples created/updated

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -14,3 +14,4 @@ Remember to include the following changes:
 - [ ] Component `README.md`
 - [ ] Tests
 - [ ] Storybook examples created/updated
+- [ ] Migration guidelines added to the description of the PR. If the guide is lengthy, consider creating a new Markdown page inside the component's docs folder and link it here
```
Comment (olliecurtis) [inline .github/pull_request_template.md:17]:
```suggestion
- [ ] For breaking changes, migration guidelines added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here
```

Comment (anambl) [reply .github/pull_request_template.md:17]: I am just thinking that it's not exactly breaking changes if we follow a deprecation cycle? I guess as a concept it is, but it won't be released as a breaking change

Comment (olliecurtis) [reply .github/pull_request_template.md:17]: I guess we can reword it to be, only adding this to provide the PR raiser context on when they should be providing migration guides otherwise it leaves it open to confusion/interpretation which might mean we have to reject a PR until we go back and ask for it

```suggestion
- [ ] For breaking changes or deprecating components/properties, migration guides added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here
```

**Review by olliecurtis — APPROVED** [review]
